Break up evaluators table (#46090)
dotnet/docs: docs/ai/conceptual/evaluation-libraries.md

## Comprehensive evaluation metrics

The evaluation libraries were built in collaboration with data science researchers from Microsoft and GitHub, and were tested on popular Microsoft Copilot experiences. The following sections show the built-in [quality](#quality-evaluators) and [safety](#safety-evaluators) evaluators and the metrics they measure.

You can also customize to add your own evaluations by implementing the <xref:Microsoft.Extensions.AI.Evaluation.IEvaluator> interface or extending the base classes such as <xref:Microsoft.Extensions.AI.Evaluation.Quality.ChatConversationEvaluator> and <xref:Microsoft.Extensions.AI.Evaluation.Quality.SingleNumericMetricEvaluator>.

### Quality evaluators

> Quality evaluators measure response quality use an LLM to perform the evaluation.

Reviewer: ```suggestion
Quality evaluators measure response quality. They use an LLM to perform the evaluation.
```

| Fluency      | Evaluates grammatical accuracy, vocabulary range, sentence complexity, and overall readability| <xref:Microsoft.Extensions.AI.Evaluation.Quality.FluencyEvaluator> |
| Coherence    | Evaluates the logical and orderly presentation of ideas | <xref:Microsoft.Extensions.AI.Evaluation.Quality.CoherenceEvaluator> |
| Equivalence  | Evaluates the similarity between the generated text and its ground truth with respect to a query | <xref:Microsoft.Extensions.AI.Evaluation.Quality.EquivalenceEvaluator> |
| Groundedness | Evaluates how well a generated response aligns with the given context | <xref:Microsoft.Extensions.AI.Evaluation.Quality.GroundednessEvaluator> |

> | Relevance (RTC), Truth (RTC), and Completeness (RTC) | Evaluates how relevant, truthful, and complete a response is | <xref:Microsoft.Extensions.AI.Evaluation.Quality.RelevanceTruthAndCompletenessEvaluator>† |

Reviewer: Could we somehow call out that RTC metrics are [Experimental]?

Author: Yes, it's called out already in the dagger footnote.

† This evaluator is marked [experimental](../../fundamentals/syslib-diagnostics/experimental-overview.md).

### Safety evaluators

Safety evaluators check for presence of harmful, inappropriate, or unsafe content in a response. They rely on the Azure AI Foundry Evaluation service, which uses a model that's fine tuned to perform evaluations.

| Metric             | Description                                                           | Evaluator type               |
|--------------------|-----------------------------------------------------------------------|------------------------------|

> | Groundedness       | Evaluates how well a generated response aligns with the given context | `GroundednessProEvaluator`   |

Reviewer: ```suggestion
| Groundedness Pro       | Uses a fine-tuned model hosted behind the Azure AI Foundry Evaluation service to evaluate how well a generated response aligns with the given context | `GroundednessProEvaluator`   |
```

> | Groundedness       | Evaluates how well a generated response aligns with the given context | `GroundednessProEvaluator`   |

Reviewer: Can we link to the docs for the evaluators in the `Evaluator type` column like you did for the quality evaluators above?

Author: Yes, let me update the API docs first so I can link to them.

| Protected material | Evaluates response for the presence of protected material             | `ProtectedMaterialEvaluator` |

> | Ungrounded human attributes | Evaluates a response for the presence of content that indicates ungrounded inference of human attributes | `UngroundedAttributesEvaluator` |

Reviewer: ```suggestion
| Ungrounded attributes | Evaluates a response for the presence of content that indicates ungrounded inference of human attributes | `UngroundedAttributesEvaluator` |
```

> | Hate content | Evaluates a response for the presence of content that's hateful or unfair | `HateAndUnfairnessEvaluator`† |

Reviewer: ```suggestion
| Hate and unfairness | Evaluates a response for the presence of content that's hateful or unfair | `HateAndUnfairnessEvaluator`† |
```

> | Self-harm content | Evaluates a response for the presence of content that indicates self harm | `SelfHarmEvaluator`† |

Reviewer: I think it may be better to remove the content suffix from the name of the metric here and below rows. That would align more with the actual names we use in the code and report.

Reviewer: Ah yes that would be a good idea. In that case, I would say use the exact names (and casing) that we use in the code (e.g. `Protected material` -> `Protected Material` etc.).
